client/kv.go

```diff
@@ -101,84 +91,44 @@ func (kv *KV) Sender() KVSender {
 	}
 }
 
-// Call invokes the KV command synchronously and returns the response
-// and error, if applicable. If preceeding calls have been made to
-// Prepare() without a call to Flush(), this call is prepared and
-// then all prepared calls are flushed.
-func (kv *KV) Call(method string, args proto.Request, reply proto.Response) error {
-	if len(kv.prepared) > 0 {
-		kv.Prepare(method, args, reply)
-		return kv.Flush()
-	}
-	if args.Header().User == "" {
-		args.Header().User = kv.User
+// Run runs the specified calls synchronously in a single batch and
+// returns any errors.
+func (kv *KV) Run(calls ...*Call) (err error) {
+	if len(calls) == 0 {
+		return nil
 	}
-	if args.Header().UserPriority == nil && kv.UserPriority != 0 {
-		args.Header().UserPriority = gogoproto.Int32(kv.UserPriority)
-	}
-	call := &Call{
-		Method: method,
-		Args:   args,
-		Reply:  reply,
-	}
-	call.resetClientCmdID(kv.clock)
-	kv.sender.Send(call)
-	err := call.Reply.Header().GoError()
-	if err != nil {
-		log.Infof("failed %s: %s", call.Method, err)
-	}
-	return err
-}
 
-// Prepare accepts a KV API call, specified by method name, arguments
-// and a reply struct. The call will be buffered locally until the
-// first call to Flush(), at which time it will be sent for execution
-// as part of a batch call. Using Prepare/Flush parallelizes queries
-// and updates and should be used where possible for efficiency.
-//
-// For clients using an HTTP sender, Prepare/Flush allows multiple
-// commands to be sent over the same connection. For transactional
-// clients, Prepare/Flush can dramatically improve efficiency by
-// compressing multiple writes into a single atomic update in the
-// event that the writes are to keys within a single range. However,
-// using Prepare/Flush alone will not guarantee atomicity. Clients
-// must use a transaction for that purpose.
-//
-// The supplied reply struct will not be valid until after a call
-// to Flush().
-func (kv *KV) Prepare(method string, args proto.Request, reply proto.Response) {
-	call := &Call{
-		Method: method,
-		Args:   args,
-		Reply:  reply,
+	// First check if any call contains an error.
+	for _, call := range calls {
+		if call.Err != nil {
```

Comment:
Why are you looking through the calls before sending them? What's the use case that would trigger an error? Could you expand comment?